src/Annotations/Field.php

```diff
@@ -4,27 +4,55 @@
 
 namespace TheCodingMachine\GraphQLite\Annotations;
 
+use Doctrine\Common\Annotations\Annotation\Attribute;
+
 /**
  * @Annotation
- * @Target({"METHOD"})
+ * @Target({"PROPERTY", "METHOD"})
  * @Attributes({
  *   @Attribute("name", type = "string"),
  *   @Attribute("outputType", type = "string"),
  *   @Attribute("prefetchMethod", type = "string"),
+ *   @Attribute("for", type = "string[]"),
+ *   @Attribute("description", type = "string"),
+ *   @Attribute("inputType", type = "string"),
  * })
  */
 class Field extends AbstractRequest
 {
     /** @var string|null */
     private $prefetchMethod;
 
+    /**
+     * Input/Output type names for which this fields should be applied to.
+     *
+     * @var string[]|null
+     */
+    private $for = null;
+
+    /**
+     * @var string|null
+     */
+    private $description;
+
+    /**
+     * @var string|null
+     */
+    private $inputType;
```

Comment:
I thought about this and came to a decision that it would be more convenient to describe fields for both type and input within one annotation. You can apply `@Input` and `@Type` annotation for the same class and in most cases you'd want to mark the same properties as fields.

With two annotations you'd need to do this twice which is kind of redundant:
```
/**
 * User entity.
 *
 * @GQL\Type()
 * @GQL\Input()
 */
class User
{
    /**
     * @var string Username or login
     *
     * @GQL\Field()
     * @GQL\InputField()
     */
    private string $username;

    /**
     * @var string User email
     *
     * @GQL\Field()
     * @GQL\InputField()
     */
    private string $email;

    /**
     * @var DateTimeInterface|null Birthday
     *
     * @GQL\Field()
     * @GQL\InputField()
     */
    private ?DateTimeInterface $birthday = null;
}
```

With one `@Field` annotation it looks more simple and clean:
```
/**
 * User entity.
 *
 * @GQL\Type()
 * @GQL\Input()
 */
class User
{
    /**
     * @var string Username or login
     *
     * @GQL\Field()
     */
    private string $username;

    /**
     * @var string User email
     *
     * @GQL\Field()
     */
    private string $email;

    /**
     * @var DateTimeInterface|null Birthday
     *
     * @GQL\Field()
     */
    private ?DateTimeInterface $birthday = null;
}
```

You need to separate input fields from output you can use `for` attribute.

But here it is the problem with `inputType` and `outputType` attributes. If I want to set the same type for both cases and the same field - which is 99% use case I think, I'd need to do this: `@GQL\Field(inputType="Date", outputType="Date")`. 

I considered alternative option - deprecate `outputType` and add another attribute `type` - which would suit for both input and output types. `@GQL\Field(type="Date")` looks much better. What do you think about that?

Reply:
Ok, I understand your will to not repeat `Field` and `InputField` annotations. It makes sense.

I'm trying to assess if there are any corner-cases that might bother us in the future.

Let's imagine this:

```
/**
 * User entity.
 *
 * @GQL\Type()
 * @GQL\Input()
 */
class User
{
    // ...

    /**
     * @var string Username or login
     *
     * @GQL\Field()
     */
    private string $password;

    public function setPassword(string $password): void {
    	$this->password = $password;
    }
}
```

In this case, we have a @Field annotation on the property, a setter, but no getter.

What should happen in your opinion? Should GraphQLite throw an error? (something like "Output field "password" is not reachable via a getter, please add a "getPassword" method). Or should we silently put only an input field and no output field?

(I'm trying to put as much attention as possible to all the error messages a user can have and to have meaningful error messages)

Reply:
It's a good question.

> Or should we silently put only an input field and no output field?
I don't think this is the way to go. At some point it might confuse developers, especially with models containing a lot of properties, why their field is not working even though there's a `@Field` annotation.

I like the idea of throwing an error. At the moment if you try accessing such field PHP will throw an error like `Cannot access private property ...`. It would be much better to display a message as you suggested to give more accurate explanation of the problem.